core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -410,6 +421,32 @@ class KRaftClusterTest {
     }
   }
 
+  private def checkReplicaManager(cluster: KafkaClusterTestKit, expectedHosting: List[(Int, List[Boolean])]): Unit = {
+    for ((brokerId, partitionsIsHosted) <- expectedHosting) {
+      val broker = cluster.brokers().get(brokerId)
+      // lock and unlock so we can read the replica manager
```

Comment:
Hmm, why do lock and unlock allow the reading of replica manager?

Reply:
I think we need to do this because `var replicaManager` in `BrokerServer` is set in a different thread by `KafkaClusterTestKit`. If my understanding of Java Memory Model is correct we need to lock/synchronize so that we force a memory barrier.

Reply:
I cleaned this up by just making `var replicaManager` volatile.